src/org/zaproxy/zap/extension/history/PopupMenuExportURLs.java

```diff
@@ -79,7 +79,19 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
         	    BufferedWriter fw = null;
                 try {
                     fw = new BufferedWriter(new FileWriter(file, isAppend));
-                    exportURLs((SiteNode) site.getModel().getRoot(), fw, html);
+                    int selectionCount=site.getSelectionCount();
+                    if (selectionCount == 0) {
+                        //No selection, export from root
+                        exportURLs((SiteNode) site.getModel().getRoot(), fw, html);
+                    } else if (selectionCount == 1) {
+                        //One selection, export starting there
+                        exportURLs((SiteNode) site.getSelectionPath()
+                            .getLastPathComponent(), fw, html);
+                    } else if (selectionCount > 1) {
+                        //Multi-selection, warn and export all
+                        extension.getView().showWarningDialog(Constant.messages.getString("exportUrls.multiselect.warning"));
+                        exportURLs((SiteNode) site.getModel().getRoot(), fw, html);

```

Comment:
The exported URIs can be added to a `Set` to avoid the duplicates (i.e. if the URI is already exported, skip it).